metronome/checkpointing/service/src/io/iohk/metronome/checkpointing/service/messages/CheckpointingMessage.scala

```diff
@@ -28,4 +28,7 @@ object CheckpointingMessage extends RPCMessageCompanion {
       state: Ledger
   ) extends CheckpointingMessage
       with Response
+
+  implicit val getStatePair =
+    pair[GetStateRequest, GetStateResponse]
```

Comment:
Interesting encoding 👍  I like the extra safety from enforcing proper response type.

Reply:
Yeah and it lets the compiler figure out the return value. Even better would be to use session types, but I went too much into the direction of queue processing.